kube/pod.go

```diff
@@ -319,20 +325,42 @@ func getEnvVars(role *model.Role, settings ExportSettings) (helm.Node, error) {
 			continue
 		}
 
-		if config.Secret {
-			secretName := "secret"
+		if config.Name == "KUBE_SECRETS_GENERATION_COUNTER" {
+			value := "1"
 			if settings.CreateHelmChart {
-				secretName += "-{{ .Release.Revision }}"
+				value = "{{ .Values.kube.secrets_generation_counter | quote }}"
 			}
+			env = append(env, helm.NewMapping("name", config.Name, "value", value))
+		}
 
-			secretKeyRef := helm.NewMapping(
-				"key", util.ConvertNameToKey(config.Name),
-				"name", secretName)
+		if config.Name == "KUBE_SECRETS_GENERATION_NAME" {
+			value := "secrets-1"
+			if settings.CreateHelmChart {
+				value = "secrets-{{ .Chart.Version }}-{{ .Values.kube.secrets_generation_counter }}"
+			}
+			env = append(env, helm.NewMapping("name", config.Name, "value", value))
+		}
 
-			envVar := helm.NewMapping("name", config.Name)
-			envVar.Add("valueFrom", helm.NewMapping("secretKeyRef", secretKeyRef))
+		if config.Secret {
+			secretKeyRef := helm.NewMapping("key", util.ConvertNameToKey(config.Name))
+			secretKeyRef.Add("name", "secrets")
 
-			env = append(env, envVar)
+			envVar := helm.NewMapping("name", config.Name, "valueFrom", helm.NewMapping("secretKeyRef", secretKeyRef))
+
+			if !settings.CreateHelmChart {
+				env = append(env, envVar)
+			} else {
+				envVar.Set(helm.Block(fmt.Sprintf("if .Values.secrets.%s", config.Name)))
+				env = append(env, envVar)
+
+				secretKeyRef := helm.NewMapping("key", util.ConvertNameToKey(config.Name))
+				secretKeyRef.Add("name", "secrets-{{ .Chart.Version }}-{{ .Values.kube.secrets_generation_counter }}")
```

Comment:
Hmm, we can't jsut add the `Block` on the `name` value, can we. Boo.

Reply:
Yeah, we can't because the name inside the mapping must be unique; the second "name" would replace the first one. :(

Reply:
I did simplify the code a bit in https://github.com/SUSE/fissile/commit/f0b082c0575eef715ec9775a3795ada7c3f4dcef though.